src/iterative_ensemble_smoother/_iterative_ensemble_smoother.py

```diff
@@ -125,32 +122,88 @@ def update_step(
+        else:
+            response_ensemble = (response_ensemble.T / observation_errors).T
 
         if ensemble_mask is None:
             ensemble_mask = np.array([True] * ensemble_size)
 
-        coefficient_matrix = update_A(
-            parameter_ensemble,
+        self.ensemble_mask = ensemble_mask
+
+        W: npt.NDArray[np.double] = create_coefficient_matrix(
             (response_ensemble - response_ensemble.mean(axis=1, keepdims=True))
-            / np.sqrt(self._initial_ensemble_size - 1),
+            / np.sqrt(ensemble_size - 1),
             R,
             E,
             D,
-            self.coefficient_matrix[ensemble_mask, :][:, ensemble_mask],
             inversion,
             truncation,
+            self.coefficient_matrix[ensemble_mask, :][:, ensemble_mask],
             step_length,
         )
 
-        self.coefficient_matrix[
-            np.outer(ensemble_mask, ensemble_mask)
-        ] = coefficient_matrix.ravel()
         self.iteration_nr += 1
-        return parameter_ensemble
+
+        self.coefficient_matrix[np.outer(ensemble_mask, ensemble_mask)] = W.ravel()
+
+    def update(self, param_ensemble: npt.NDArray[np.double]) -> npt.NDArray[np.double]:
+        # Line 9 of Algorithm 1
+        ensemble_size = self.ensemble_mask.sum()
+        transition_matrix: npt.NDArray[np.double] = np.identity(
+            ensemble_size
+        ) + self.coefficient_matrix[self.ensemble_mask, :][
+            :, self.ensemble_mask
+        ] / np.sqrt(
+            ensemble_size - 1
+        )
+        return param_ensemble @ transition_matrix
+
+    def __repr__(self) -> str:
+        return (
+            f"SIES(ensemble_size={self._initial_ensemble_size}, "
+            f"max_steplength={self.max_steplength}, min_steplength={self.min_steplength}, "
+            f"dec_steplength={self.dec_steplength})"
+        )
+
+
+class ES:
+    def __init__(self) -> None:
+        self.smoother: Optional[SIES] = None
+
+    def fit(
+        self,
+        response_ensemble: npt.NDArray[np.double],
+        observation_errors: npt.NDArray[np.double],
+        observation_values: npt.NDArray[np.double],
+        *,
+        noise: Optional[npt.NDArray[np.double]] = None,
+        truncation: float = 0.98,
+        inversion: InversionType = InversionType.EXACT,
+        param_ensemble: Optional[npt.NDArray[np.double]] = None,
+    ) -> None:
+        self.smoother = SIES(ensemble_size=response_ensemble.shape[1])
+        self.smoother.fit(
+            response_ensemble,
+            observation_errors,
+            observation_values,
+            noise=noise,
+            truncation=truncation,
+            step_length=1.0,
+            ensemble_mask=None,
+            inversion=inversion,
+            param_ensemble=param_ensemble,
+        )
+
+    def update(self, param_ensemble: npt.NDArray[np.double]) -> npt.NDArray[np.double]:
+        assert self.smoother is not None
+        return self.smoother.update(param_ensemble)
```

Comment:
This looks correct to me. I did not remember if one iteration of SIES with step length 1.0 (as implemented here) was identical to ES, so I compared both the reference book on ES (Evensen 2009, section 9.5) with Algorithm 1 of Evensen 2019.

In the first iteration and without projection, with step-size 1.0, the update in Line 9 of Algorithm 1 can be written
`X_posterior = X_prior @ (I + Y.T @ np.linalg.inv(Y@Y.T + E@E.T ) @ H`
Which should equal Equation 9.27 of Evensen 2009.

Thus all of this should be okay! 👍

Reply:
But please someone else also double check this, @dafeda I think this is what we talked about, that you have checked, and it is evidently what is implemented.

Should it be commented somewhere that this is the case?